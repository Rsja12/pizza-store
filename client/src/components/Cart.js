import React from 'react';
import { connect } from 'react-redux';

import CartTable from './CartTable';
import Pricing from './Pricing';
import '../styles/Cart.css';

export const Cart = ({ cart }) => {
    const renderCart = () => {
        if (cart.length === 0) {
            return <div className='cart-msg'>Your cart is currently empty</div>;
        } else {
            return <CartTable cart={cart} />;
        }
    };

    return (
        <div>
            <div className='cart-container'>{renderCart()}</div>
            <Pricing cart={cart} />
        </div>
    );
};

const mapStateToProps = (state) => ({
    cart: Object.values(state.cart),
});

export default connect(mapStateToProps)(Cart);
