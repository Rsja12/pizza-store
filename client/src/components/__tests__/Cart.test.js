import React from 'react';
import { shallow } from 'enzyme';

import { Cart } from '../Cart';

it ('renders when there are items in cart', () => {
    const mockProp = [{
        id: 1,
        name: 'pepperoni',
        quantity: 2
    }];

    expect(shallow(<Cart cart={mockProp} />)).toMatchSnapshot();
});

it ('renders when cart is empty', () => {
    const mockProp = [];

    expect(shallow(<Cart cart={mockProp} />)).toMatchSnapshot();
});