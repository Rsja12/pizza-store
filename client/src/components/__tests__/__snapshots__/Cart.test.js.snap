// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when cart is empty 1`] = `ShallowWrapper {}`;

exports[`renders when there are items in cart 1`] = `ShallowWrapper {}`;
